sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/Artifact.java

```diff
@@ -141,9 +149,13 @@ public final SortedSet<Version> getVersions() {
   }
 
   public final SortedSet<Version> getPublicVersions() {
+    return getPublicVersions(null);
+  }
+
+  public final SortedSet<Version> getPublicVersions(@Nullable Product product) {
     SortedSet<Version> versions = new TreeSet<>();
     for (Release release : releases) {
-      if (release.isPublic()) {
+      if (release.getProduct() == product && release.isPublic()) {
```

Comment:
Nitpick: maybe .equals() instead of ==

Reply:
Hmm, I prefer to use == for enums. It is faster and null safe.